temporalio/worker/_workflow_instance.py

```diff
@@ -642,11 +640,13 @@ async def run_query() -> None:
                         raise ValueError(
                             f"Expected 1 result payload, got {len(result_payloads)}"
                         )
-                    command.respond_to_query.succeeded.response.CopyFrom(
-                        result_payloads[0]
-                    )
+                command = self._add_command()
+                command.respond_to_query.query_id = job.query_id
+                command.respond_to_query.succeeded.response.CopyFrom(result_payloads[0])
```

Comment:
Non-blocking: in general I believe `try` blocks should enclose as little as possible. it seems that these 3 lines could go after?

Reply:
No because this needs to only be called on the success path, there is a failure path that does another thing, there is no need for any common code after the try